client/rest-high-level/src/main/java/org/elasticsearch/client/indices/PutIndexTemplateRequest.java

```diff
@@ -278,18 +278,14 @@ public PutIndexTemplateRequest source(Map<String, Object> templateSource) {
         Map<String, Object> source = templateSource;
         for (Map.Entry<String, Object> entry : source.entrySet()) {
             String name = entry.getKey();
-            if (name.equals("template")) {
-                if(entry.getValue() instanceof String) {
-                    patterns(Collections.singletonList((String) entry.getValue()));
-                }
-            } else if (name.equals("index_patterns")) {
+            if (name.equals("index_patterns")) {
```

Comment:
This is a hard break -- users who specified `template` in the source would not have seen a deprecation warning about it, since we convert it to `index_patterns` before sending the REST request to the server. I wasn't sure how to address this, any guidance here would be great.

Reply:
We can and should add a check to the Deprecation Info API, but that still leaves folks using the OSS distribution lacking a notification.

Reply:
To help clarify my original comment, I am wondering how to warn users of the Java HLRC that they can no longer specify `template` when using `PutIndexTemplateRequest#source`. Before this PR we would silently convert it to `index_patterns` before sending it off to the server.

Reply:
Could we change the 7.x branch to add a deprecation warning if `template` was used and remove it in 8.0? That seems reasonable to me (if possible with the HLRC)

Reply:
@dakrone this is a nice idea, I'll open a follow-up PR against 7.x to emit a deprecation warning if `template` is used.